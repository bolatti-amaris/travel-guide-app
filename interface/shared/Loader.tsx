import React from 'react';
import { View, Text, ActivityIndicator } from 'react-native';

interface LoaderProps {
  message: string;
}

const Loader = ({ message } : LoaderProps) => {
  return (
    <View className="flex-1 justify-center items-center">
      <ActivityIndicator size="large" className="mb-4" />
      <Text className="text-base text-gray-500 font-work-light text-center">{message}</Text>
    </View>
  );
};

export default Loader;