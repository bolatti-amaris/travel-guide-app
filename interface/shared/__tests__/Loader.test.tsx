import React from "react";
import { render } from "@testing-library/react-native";
import Loader from "../Loader";

describe("Loader", () => {
  it("renders spinner and message correctly", () => {
    const { getByTestId } = render(<Loader message="Loading data..." />);

    const wrapper = getByTestId("loader-wrapper");
    const spinner = getByTestId("loader-spinner");
    const message = getByTestId("loader-message");

    expect(wrapper).toBeTruthy();
    expect(spinner).toBeTruthy();
    expect(message.props.children).toBe("Loading data...");
  });

  it("applies correct classNames to wrapper and message", () => {
    const { getByTestId } = render(<Loader message="Verifying..." />);

    expect(getByTestId("loader-wrapper").props.className).toContain("flex-1");
    expect(getByTestId("loader-wrapper").props.className).toContain(
      "bg-background",
    );
    expect(getByTestId("loader-message").props.className).toContain(
      "text-icon",
    );
    expect(getByTestId("loader-message").props.className).toContain(
      "text-center",
    );
  });

  it("matches snapshot", () => {
    const tree = render(<Loader message="Please wait..." />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
