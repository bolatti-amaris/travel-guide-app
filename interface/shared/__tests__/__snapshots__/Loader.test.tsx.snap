// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<View
  className="flex-1 justify-center items-center bg-background dark:bg-background-dark"
  testID="loader-wrapper"
>
  <ActivityIndicator
    className="mb-4"
    size="large"
    testID="loader-spinner"
  />
  <Text
    className="text-base text-icon dark:text-icon-dark font-work-light text-center"
    testID="loader-message"
  >
    Please wait...
  </Text>
</View>
`;
